api/queries_issue.go

```diff
@@ -246,8 +247,18 @@ func IssueList(client *Client, repo ghrepo.Interface, state string, labels []str
 	if mentionString != "" {
 		variables["mention"] = mentionString
 	}
+
 	if milestoneString != "" {
-		variables["milestone"] = milestoneString
+		milestones, err := RepoMilestonesREST(client, repo)
```

Comment:
You can use our existing GraphQL-based `RepoMilestones()` function and read the GraphQL ID for a milestone. Then, you can base64-decode the GraphQL ID, which will get you a string like `09:Milestone12345`. The final number in that string is the database ID of the milestone

Reply:
Good point :+1: 
Should we assume the the decoded string will follow a pattern like: `^\d+:Milestone(\d+)$`?

Reply:
Yes, I think you can safely assume that and let it error out if it doesn't.